### PR TITLE
Add _pg_expandarray table function

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits

This is required for https://github.com/crate/crate/issues/9566

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/expression/tablefunctions/PgExpandArray.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    public static void register(TableFunctionModule module) {
+        module.register(
+            FUNCTION_NAME,
+            new BaseFunctionResolver(FuncParams.builder(Param.ANY_ARRAY).build()) {
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> types) throws IllegalArgumentException {
+                    assert types.size() == 1 : "_pg_expandarray must have a single argument due to funcParams";
+                    DataType<?> dataType = types.get(0);
+                    assert dataType instanceof ArrayType : "Argument to _pg_expandarray must be an array";
```
Comment (seut) [inline sql/src/main/java/io/crate/expression/tablefunctions/PgExpandArray.java]: Aren't these assertions already ensured by the FuncParams definition?

File: sql/src/main/java/io/crate/expression/scalar/SubscriptFunctions.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                }
+                Function recordSubscript = new Function(
+                    new FunctionInfo(
+                        new FunctionIdent(SubscriptRecordFunction.NAME, List.of(baseType, DataTypes.STRING)),
+                        rowType.getFieldType(idx)
+                    ),
+                    List.of(baseSymbol, Literal.of(child))
+                );
+                if (path.size() > 1) {
+                    return tryCreateSubscript(recordSubscript, path.subList(1, path.size() - 1));
```
Comment (seut) [inline sql/src/main/java/io/crate/expression/scalar/SubscriptFunctions.java]: Is this covered by any test already?

File: sql/src/main/java/io/crate/expression/tablefunctions/PgExpandArray.java (hunk 1 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/expression/tablefunctions/PgExpandArray.java]: Yes they are, the assertion message earlier also says so: `"_pg_expandarray must have a single argument due to funcParams";`

I like to add those to clarify that there is some kind of contract in place.

File: sql/src/main/java/io/crate/expression/scalar/SubscriptFunctions.java (hunk 2 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/expression/scalar/SubscriptFunctions.java]: No. Good point, I'll add some.

**Comment by mfussenegger** [comment]
> the commit message here 56f1761 states that no documentation is added but the commit contains the documentation. Do I miss something? otherwise maybe update the commit message.

Strange, I thought I had updated it. Thanks for catching.

**Pull request merged**